CHAOS-6966 Pod JVM modify return (#8807)
harness/developer-hub: docs/chaos-engineering/use-harness-ce/chaos-faults/kubernetes/pod/pod-jvm-modify-return.md

---
id: pod-jvm-modify-return
title: Pod JVM modify return
---

Pod JVM modify return modifies the return value of a method in a Java application executing on a Kubernetes pod, for a specific duration.

:::tip
JVM chaos faults use the [Byteman utility](https://byteman.jboss.org/) to inject chaos faults into the JVM.
:::

![Pod JVM modify return](./static/images/pod-jvm-modify-return.png)

## Use cases
Pod JVM modify return:
- Helps test the functionality of snippets of code by replacing specific portions of the request or response body to simulate different scenarios and validate how your application handles different data variations.
- Helps obscure or redact personally identifiable information (PII), such as email addresses or phone numbers, before logging or transmitting the data for security and privacy compliance.
- Determines how efficiently an application recovers and returns to normalcy.

### Mandatory tunables
<table>
  <tr>
    <th> Tunable </th>
    <th> Description </th>
    <th> Notes </th>
  </tr>
<tr>
    <td> CLASS </td>

>     <td> Specify as <b>packageName.className</b> that specifies the class in which you define the exception. </td>

Reviewer: ```suggestion
    <td> Specify as <b>packageName.className</b> that specifies the class in which you define the target method. </td>
```

    <td> For example, <code>org.framework.appName.system.WelcomeController</code>. </td>
  </tr>
  <tr>
    <td> METHOD </td>

>     <td> The method to which exception is applied. </td>

Reviewer: ```suggestion
    <td> The method whose return value is modified. </td>
```
